smartsim/_core/utils/serialize.py

```diff
@@ -0,0 +1,237 @@
+    manifest_file = manifest_dir / "manifest.json"
+
+    run_id = _helpers.create_short_id_str()
+    telemetry_data_root = manifest_dir / f"{manifest.metadata.exp_name}/{run_id}"
+
+    new_run = {
+        "run_id": run_id,
+        "timestamp": int(time.time_ns()),
+        "model": [
+            _dictify_model(
+                model,
+                *telemetry_metadata,
+                telemetry_data_root / "model",
+            )
+            for model, telemetry_metadata in manifest.models
+        ],
+        "orchestrator": [
+            _dictify_db(db, nodes_info, telemetry_data_root / "database")
+            for db, nodes_info in manifest.databases
+        ],
+        "ensemble": [
+            _dictify_ensemble(ens, member_info, telemetry_data_root / "ensemble")
+            for ens, member_info in manifest.ensembles
+        ],
+    }
+    try:
+        with open(manifest_file, "r", encoding="utf-8") as file:
+            manifest_dict = json.load(file)
+    except (FileNotFoundError, json.JSONDecodeError):
+        manifest_dict = {
+            "schema info": {
+                "schema_name": "entity manifest",
+                "version": "0.0.1",
+            },
+            "experiment": {
+                "name": manifest.metadata.exp_name,
+                "path": manifest.metadata.exp_path,
+                "launcher": manifest.metadata.launcher_name,
+            },
+            "runs": [new_run],
+        }
+    else:
+        manifest_dict["runs"].append(new_run)
+    finally:
+        with open(manifest_file, "w", encoding="utf-8") as file:
+            json.dump(manifest_dict, file, indent=2)
+
+
+def _dictify_model(
+    model: Model,
+    step_id: t.Optional[str],
+    task_id: t.Optional[str],
+    managed: t.Optional[bool],
+    out_file: str,
+    err_file: str,
+    telemetry_data_path: Path,
+) -> t.Dict[str, t.Any]:
+    colo_settings = (model.run_settings.colocated_db_settings or {}).copy()
+    db_scripts = t.cast("t.List[DBScript]", colo_settings.pop("db_scripts", []))
+    db_models = t.cast("t.List[DBModel]", colo_settings.pop("db_models", []))
+    return {
+        "name": model.name,
+        "path": model.path,
+        "exe_args": model.run_settings.exe_args,
+        "run_settings": _dictify_run_settings(model.run_settings),
+        "batch_settings": _dictify_batch_settings(model.batch_settings)
+        if model.batch_settings
+        else {},
+        "params": model.params,
+        "files": {
+            "Symlink": model.files.link,
+            "Configure": model.files.tagged,
+            "Copy": model.files.copy,
+        }
+        if model.files
+        else {
+            "Symlink": [],
+            "Configure": [],
+            "Copy": [],
+        },
+        "colocated_db": {
+            "settings": colo_settings,
+            "scripts": [
+                {
+                    script.name: {
+                        "backend": "TORCH",
```

Comment:
is this always torch?

Reply:
We resolved this offline, but wanted to make sure is is documented here for any future readers!

At time of writing `DBScript` at the moment only refers to TorchScripts, and as such the `DBScript` has no `backend` attr because we assume  that it must be `TORCH`. This may change in the future, but for now it is the current state of the world.